File: jekyun-park/week2/programmers/49993.swift
//
//  main.swift
//  SwiftAlgorithms
//
//  Created by 박제균 on 2022/01/06.
//

import Foundation


func solution(_ skill:String, _ skill_trees:[String]) -> Int {
    var answer = 0
    
    for tree in skill_trees {
        let list = tree.filter{ skill.contains($0) }
        let prefix = skill.prefix(list.count)
        print(list,prefix)
        if prefix == list { answer += 1}
    }
    
    return answer
}

print(solution("CBD", ["BACDE", "CBADF", "AECB", "BDA"]))

File: JaeYoung/week02/49993.py
def solution(skill, skill_trees):
    
    answer = len(skill_trees)
    
    for tree in skill_trees:
        tmp_skill = skill
        for x in tree:
            if x in tmp_skill: #배울 스킬이 선행스킬에 있고
                if tmp_skill[0] == x: #선행으로 배워야할 스킬이면 tmp_skill 의 해당 스킬을 삭제
                    tmp_skill = tmp_skill.replace(x, "")
                else: #선행으로 배워야할 스킬이 아니면 answer에 -1 해준다
                    answer -= 1
                    break

    return answer